api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -0,0 +1,179 @@
+                    f"of release {release}")
+
+        # general check from pkg manager point of view
+        try:
+            self._check_repo_is_valid(release, f"sw_upgrade_{repo_name}")
+        except SaltCmdResultError as exc:
+            raise SWUpdateRepoSourceError(
+                                        repo_name, f"malformed repo: '{exc}'")
+
+        # there is no the same release repo is already active
+        if self._is_repo_enabled(f'sw_update_{release}'):
+            err_msg = (
+                "SW update repository for the release "
+                f"'{release}' has been already enabled"
+            )
+            logger.warning(err_msg)
+
+            # TODO IMPROVE later raise and error
+            if False:
+                raise SWUpdateRepoSourceError(
+                    str(repo.source),
+                    (
+                        f"SW update repository for the release "
+                        f"'{release}' has been already enabled"
+                    )
+                )
+
+    def dynamic_validation(self, params: inputs.SWUpgradeRepo, targets: str):  # noqa: C901, E501
+        """
+        Validate single SW upgrade ISO structure.
+
+        Parameters
+        ----------
+        params: inputs.SWUpgradeRepo
+            Input repository parameters
+        targets: str
+            Salt target to perform base mount and validation logic
+
+        Returns
+        -------
+
+        """
+        repo = params
+
+        if repo.is_special():
+            logger.info("Skipping update repo validation for special value: "
+                        f"{repo.source}")
+            return
+
+        logger.info(f"Validating upgrade repo: release {repo.release}, "
+                    f"source {repo.source}")
+
+        candidate_repo = inputs.SWUpgradeRepo(REPO_CANDIDATE_NAME, repo.source)
+        # TODO IMPROVE VALIDATION EOS-14350
+        #   - there is no other candidate that is being verified:
+        #     if found makes sense to raise an error in case the other
+        #     logic is still running, if not - forcibly remove the previous
+        #     candidate
+        #   - after first mount 'sw_update_candidate' listed in disabled repos
+        # TODO: need to have logic for sw upgrade
+        # if self._does_repo_exist(f'sw_update_{candidate_repo.release}'):
+        #     logger.warning(
+        #       'other repo candidate was found, proceeding with force removal'
+        #     )
+        # TODO IMPROVE: it is not enough it may lead to locks when
+        #  provisioner doesn't unmount `sw_update_candidate` repo
+        # raise SWUpdateError(reason="Other repo candidate was found")
+
+        try:
+            logger.debug("Configuring upgrade candidate repo for validation")
+            self._prepare_repo_for_apply(candidate_repo, enabled=False)
+
+            super(SetSWUpdateRepo, self)._run(candidate_repo, targets)
```

Comment:
Yes, it was used to call the `_run` method of `Set` parent class instead of `_run` method of `SetSWUpdateRepo` parent.
I changed the structure of API for `Set` class to avoid usage of `super` function.